packages/nimble-components/src/table-column/mixins/groupable-column.ts

```diff
@@ -0,0 +1,68 @@
+import {
+    attr,
+    nullableNumberConverter,
+    observable
+} from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+type GroupableTableColumn = Pick<
+TableColumn,
+| 'internalGroupingDisabled'
+| 'internalGroupIndex'
+| 'internalGroupHeaderViewTag'
+>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type GroupableTableColumnConstructor = abstract new (...args: any[]) => GroupableTableColumn;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinGroupableColumnAPI<
+    TBase extends GroupableTableColumnConstructor
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow grouping
+     * by the values in that column.
+     */
+    abstract class GroupableColumn extends base {
+        public groupingDisabled = false;
+
+        public groupIndex?: number | null = null;
+
+        public abstract groupHeaderViewTag: string;
```

Comment:
I'm not sure I agree with your suggestion. While you are correct that this API is not meant to be overridden by end users, this can be solved in the same way it is for the [`cellViewTag` property of the `TableColumnText` class] (https://github.com/ni/nimble/pull/1138/files#diff-5058c96c2bc55c407497e6c95b298d71299edba7119549c353cd6985500f135d) in @msmithNI's PR, by simply marking the property as `readonly`.

Additionally, I'm not sure it makes sense for this property _not_ to be on the mixin. It's basically enforcing that a column implementation that decides it allows grouping, _must_ provide the visual for the group header. Another missing aspect of the current implementation could be simply marking the `abstract` property on the mixin as `internal` just as the abstract `cellViewTag` property will be marked on `TableColumn`.

Reply:
I think I'm in favor of using Malcolm's pattern of marking it `abstract readonly cellViewTag: string;` in TableColumn.

Having two different properties and observers and change handlers to deal with what is ultimately a constant the plugin author should provide seems kinda complex.